src/components/PrintModal/WatermarkModal/WatermarkModal.js

```diff
@@ -77,28 +77,26 @@ class WatermarkModal extends React.PureComponent {
       isColorPaletteVisible: false,
       locationSettings,
       previousLocationSettings: locationSettings,
+      preExistingWatermark: {},
     };
     this.canvasContainerRef = React.createRef();
 
-    this.handleWatermarkRenderFxn = () => {
+    this.handleWatermarkRenderFxn = async () => {
       if (this.props.isVisible) {
+        const preExistingWatermark = await core.getWatermark();
```

Comment:
store existing watermarks (if any) before we add in the watermarks created by watermark modal. when dialog closes, put back in pre-existing watermarks.

Reply:
Why is `preExistingWatermark ` on state? Do you use it to rerender the view?

Also, it seems a bit weird that we define a function in the constructor. Could you tell me the reason why you are doing this?

Reply:
Hmm if we need to use it somewhere else in this component in the future then we can store it on `this` instead of `this.states`... My opinion is that properties of `this.states` should be related to how the component look like. If a variable has nothing to do with the view then we should store it on `this`, or even locally...

> I needed to define a function so that I can get its reference to that I can add/remove the event listener

I think if you define it like the rest of the functions(for example: resetForm), you can still access it via `this`?... So I think you can just define it like others. What do you think?

Reply:
`My opinion is that properties of this.states should be related to how the component look like. If a variable has nothing to do with the view then we should store it on this, or even locally`
It seems I lacked a bit of understanding of `this.states` vs `this`. I cross checked what you were saying on Google as well. **I will move it out of the state and store it locally or on `this`**.

`I think if you define it like the rest of the functions(for example: resetForm), you can still access it via this?... So I think you can just define it like others`
**I'm fine with that suggestion. By doing it that way, I think it will match the consistency of how the other functions are structured.**

Reply:
addressed